src/components/TodoApp.tsx

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState } from 'react';
+
+import { Todo } from '../types/Todo';
+import { Notification } from '../types/Notification';
+import { Status } from '../types/Status';
+
+import * as TodosService from '../api/todos';
+
+import { UserWarning } from '../UserWarning';
+import { TodoFooter } from './TodoFoter';
+import { TodoList } from './TodoList';
+import { TodoNotificaition } from './TodoNotification';
+import { TodoHeader } from './TodoHeader';
+import { getVisibleTodos } from '../service/filter';
+
+const USER_ID = 11293;
+
+export const TodoApp = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    TodosService.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(Notification.load);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, [errorMessage]);
+
+  const addTodo = ({ title, userId, completed }: Todo) => {
+    setErrorMessage('');
+
+    return TodosService.createTodos({ title, userId, completed })
+      .then(newTodos => {
+        setTodos(currentTodo => [...currentTodo, newTodos]);
+      })
+      .catch((error) => {
+        setErrorMessage(Notification.add);
+        throw error;
+      });
+  };
+
+  const deleteTodo = (todoId: number) => {
+    TodosService.deleteTodo(todoId);
+    setTodos(currentTodo => currentTodo.filter(todo => todo.id !== todoId));
```

Comment:
change state only after receive response from server amd handle error in catch
check other similer cases.

```suggestion
    TodosService.deleteTodo(todoId).then(() => {
         setTodos(currentTodo => currentTodo.filter(todo => todo.id !== todoId));
    }).catch( // handle error);
```